tests/pipeline/BadBehaviorLibrary.cpp

```diff
@@ -0,0 +1,62 @@
+/// \file BadBehaviorLibrary.cpp
+/// \brief This file, when compiled into a library, crashes in a manner
+/// configured in the environment variables.
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <csignal>
+#include <cstdio>
+#include <cstdlib>
+#include <iostream>
+#include <string>
+#include <string_view>
+
+int *SomeGlobalPointerNobodyWillInitialize;
+
+#ifdef __clang__
+#define OPTNONE [[clang::optnone]]
+#else
+#define OPTNONE __attribute__((optimize("O0")))
+#endif
+
+static void printAndFlush(std::string_view String) {
+  std::cout << String << std::endl;
+  std::cout.flush();
+}
+
+OPTNONE static void doCrash() {
```

Comment:
Why optnone?

Reply:
Yes. See the implementation.

Reply:
I'm confused. O thought you had `*((int *) nullptr) = 69` which could be DCE'd away since it's UB. But looking at your code, it all seems legit. Are you sure you need `OPTNONE`?
Out of curiosity, does `__attribute__((optimize("O0")))` work on `clang`?

Reply:
> Yes, clang does optimize out that write: https://godbolt.org/z/79hbsPra4 gcc does not.

That's because there's `__builtin_unreachable`, if you drop it, it won't drop the store.

Reply:
Makes sense. Updated.